### PR TITLE
Cascade

### PR DESCRIPTION
**Pull request by dikshayp**
Cascading delete for all the tables.
Renamed TKReader to TKReader1

### CODE REVIEW

File: src/main/java/com/asu/seatr/handlers/StudentTaskHandler.java [hunk 1]
```diff
@@ -109,6 +121,45 @@ public static void delete(StudentTask studentTask)
 		session.getTransaction().commit();
 		session.close();
 	}
+	/*
```
Comment (yeskarthik) [inline src/main/java/com/asu/seatr/handlers/StudentTaskHandler.java]: do you want to delete the commented code?

Comment (dikshayp) [inline src/main/java/com/asu/seatr/handlers/StudentTaskHandler.java]: Yea, now we donot need that

File: src/main/java/com/asu/seatr/handlers/CourseHandler.java [hunk 2]
```diff
@@ -72,4 +74,53 @@ public static void updateCourseByExternalID(String external_id,Course course){
...
+		if(courseIdList == null || courseIdList.isEmpty())
+		{
+			return null;
+		}
+		try
+		{
+		SessionFactory sf = HibernateUtil.getSessionFactory();
+		Session session = sf.openSession();
+		session.beginTransaction();
+		String hql = "select c.id from Course c where c.external_id in :courseIdList";
```
Comment (yeskarthik) [inline src/main/java/com/asu/seatr/handlers/CourseHandler.java:130]: why not use the Criteria way?

Comment (dikshayp) [inline src/main/java/com/asu/seatr/handlers/CourseHandler.java:130]: Because, we are not using single equality condition. we are checking the whole list

**Comment by yeskarthik** [comment]
In KCAPITest.java the batchsave method is throwing compilation error.

File: src/main/java/com/asu/seatr/handlers/CourseHandler.java (hunk 2 above)
Comment (yeskarthik) [inline src/main/java/com/asu/seatr/handlers/CourseHandler.java:130]: Okay, we'll try to see if we can do the criteria way or hibernate way for such a thing later.

**Comment by dikshayp** [comment]
KCAPITest fixed. Please Review

**Comment by yeskarthik** [comment]
I'm getting a ConstraintViolationException when I send the smae request twice. Can you check?

**Comment by yeskarthik** [comment]
:+1: except for one small bug that @dikshay mentioned.

**Comment by dikshayp** [comment]
@lzhang90  When i submit the first task_response, the task that does not belong to the  said course, also got inserted into the recomm table. So a student got a recommended task, from a course that he has not taken.

**Pull request merged**